src/providers/application/ApplicationProvider.tsx

```diff
@@ -238,13 +238,24 @@ function ApplicationProvider({ children }: ApplicationProviderProps) {
   }
 
   async function submitApplication() {
+    dispatch({ type: ApplicationActionType.LOADING });
     const response = await fetch(
       `/api/applications/${application!.id}/submit`,
       {
         method: "POST",
       },
     );
-    if (response.ok) fetchApplication();
+    if (response.ok) {
+      fetchApplication();
+    } else {
+      const responseJson = await response.json();
+      const message = responseJson.title || "Failed to submit application";
```

Comment:
important: it's better to present `error.detail`, instead of `error.title` - Title it's just a human readable version of the error type, but it gives no detail on what exactly has happened. 

https://github.com/GenomicDataInfrastructure/gdi-userportal-access-management-service/blob/202dd151ea67256f6122b71e91e9cb3a6d3c4bf3/src/main/openapi/daam.yaml#L950